packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,224 @@
+| defaultChecked      | `boolean`  | No       | false     | default state of the checkbox, if isChecked is not provided the checkbox will become [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html)                                                            |
+| onChange            | `({ event, isChecked, value }) => void` | No       | undefined | The function to be called when the checkbox state changes.                                                                                               |
+| isIndeterminate     | `boolean`  | No       | false     | mixed state of the checkbox                                                                                                                              |
+| isDisabled          | `boolean`  | No       | `false`   | Control whether the checkbox is disabled or not.                                                                                                         |
+| isRequired          | `boolean`  | No       | `false`   | Control whether the checkbox is required or not.                                                                                                         |
+| helpText          | `string`   | No       | undefined | The helper text to be rendered                                                                                                                           |
+| errorText          | `string`   | No       | undefined | The error text to be rendered                                                                                                                           |
+| validationState           | `undefined \| 'error'`  | No       | `undefined`   | Control whether the checkbox is invalid or not.                                                                                                          |
+| name (html native)  | `string`   | No       | undefined | The name of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)            |
+| value (html native) | `string`   | No       | undefined | The value of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value) |
+
+### Examples:
+
+#### Basic
+
+```tsx
+// Label text
+<Checkbox>is blade awesome?</Checkbox>
+// Helper text
+<Checkbox helpText="Hint: yes">is blade awesome?</Checkbox>
+// uncontrolled
+<Checkbox defaultChecked={true}>is blade awesome?</Checkbox>
+// controlled
+<Checkbox isChecked={true | false} onChange={({ isChecked }) => {}}>is blade awesome?</Checkbox>
+```
+
+#### Validations
+
+And `validationState` will pass `aria-invalid` attributes to indicate invalid state.
+
+```tsx
+<Checkbox validationState={'error' | undefined}>is blade awesome?</Checkbox>
+```
+
+## `CheckboxGroup` API
```

Comment:
no `isRequired` prop on checkboxgroup? that shall be a general validation for all input fields. thoughts?

Reply:
Nope. no required prop on checkbox. Passing down required prop from checkboxgroup to individual checkboxes doesn't make sense since we don't know if user wants all/some or atleast one to be required or not. 

We only have neccessityIndicator=required